src/wp-includes/block-template-utils.php

```diff
@@ -1444,36 +1444,78 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
  * @since 6.5.0
  * @access private
  *
- * @param stdClass        $post    An object representing a template or template part
+ * @param stdClass        $changes An object representing a template or template part
  *                                 prepared for inserting or updating the database.
  * @param WP_REST_Request $request Request object.
  * @return stdClass The updated object representing a template or template part.
  */
-function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
+function inject_ignored_hooked_blocks_metadata_attributes( $changes, $request ) {
 	$filter_name = current_filter();
 	if ( ! str_starts_with( $filter_name, 'rest_pre_insert_' ) ) {
-		return $post;
+		return $changes;
 	}
 	$post_type = str_replace( 'rest_pre_insert_', '', $filter_name );
 
 	$hooked_blocks = get_hooked_blocks();
 	if ( empty( $hooked_blocks ) && ! has_filter( 'hooked_block_types' ) ) {
-		return $post;
+		return $changes;
 	}
 
-	// At this point, the post has already been created.
 	// We need to build the corresponding `WP_Block_Template` object as context argument for the visitor.
 	// To that end, we need to suppress hooked blocks from getting inserted into the template.
 	add_filter( 'hooked_block_types', '__return_empty_array', 99999, 0 );
-	$template = $request['id'] ? get_block_template( $request['id'], $post_type ) : null;
+
+	if ( isset( $changes->ID ) ) {
+		$post = get_post( $changes->ID );
+	} else {
+		// This means that there's not post for this template in the DB yet.
+		$post = new stdClass; // Is this correct? Might instead want to init a WP_Post with some fields set.
```

Comment:
In the post editor it’s being set to the post object with some initial data. Let me find the code.

Reply:
I think this is what gets called to create a new draft post:

https://github.com/WordPress/wordpress-develop/blob/3d3d5106971ae78845dfbd2983fc1c356f0d5a4e/src/wp-admin/post-new.php#L66

I was thinking about the following logic that is rather unrelated:

https://github.com/WordPress/wordpress-develop/blob/3d3d5106971ae78845dfbd2983fc1c356f0d5a4e/src/wp-admin/edit-form-blocks.php#L91-L105

I don't know if the `get_default_post_to_edit` is a good way to go here.